src/Codeception/Configuration.php

```diff
@@ -628,9 +627,8 @@ public static function append(array $config = []): array
     /**
      * @param array|bool|null $a1
      * @param array|bool|null $a2
-     * @return array|bool
      */
-    public static function mergeConfigs($a1, $a2)
+    public static function mergeConfigs($a1, $a2): array|bool|null|string
```

Comment:
It looks like `string` type originated from `$res[$k2] = self::mergeConfigs($a1[$k2], $v2);` in line 658.

I think that it is the only place when the first parameter of `mergeConfig` can be not array.
But return type above is not sufficient, element value can be int and float too,
so we have to go full `mixed`, or make sure that `mergeConfig is not called if `$a1[$k2]` is not array.

```php
            if (!isset($a1[$k2]) || !is_array($a1[$k2]) {
                $res[$k2] = $v2;
                unset($a1[$k2]);
                continue;
            }

            $res[$k2] = self::mergeConfigs($a1[$k2], $v2);
```

If this change is made, we can set return type to `array`.